hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java

```diff
@@ -1139,6 +1151,20 @@ public void setCertClient(CertificateClient newClient) throws IOException {
     serviceInfo = new ServiceInfoProvider(secConfig, this, certClient);
   }
 
+  /**
+   * For testing purpose only. This allows setting up ScmBlockLocationClient
+   * without having to fully setup a working cluster.
+   */
+  @VisibleForTesting
+  public void setScmTopologyClient(
+      ScmTopologyClient scmTopologyClient) {
+    this.scmTopologyClient = scmTopologyClient;
+  }
+
+  public NetworkTopology getClusterMap() {
+    return clusterMap;
```

Comment:
When does clusterMap need to be refreshed?

Reply:
Currently, we initialize the `clusterMap` (type: `NetworkTopology`) in OM during startup based on the schema file and the fetched `clusterTree` information (type: `InnerNode`) from `ScmTopologyClient` (which implements a background thread responsible for periodic caching and refreshing of this `clusterTree` obtained from SCM at every ~1 hr interval).